src/fios.cpp

```diff
@@ -554,6 +559,42 @@ void FiosGetHeightmapList(SaveLoadOperation fop, bool show_dirs, FileList &file_
 	FiosGetFileList(fop, show_dirs, &FiosGetHeightmapListCallback, subdir, file_list);
 }
 
+/**
+ * Callback for FiosGetTownDataList.
+ * @param fop Purpose of collecting the list.
+ * @param file Name of the file to check.
+ * @return a FIOS_TYPE_JSON type of the found file, FIOS_TYPE_INVALID if not a valid JSON file, and the title of the file (if any).
+ */
+static std::tuple<FiosType, std::string> FiosGetTownDataListCallback(SaveLoadOperation fop, const std::string &file, const std::string_view ext)
+{
+	if (fop == SLO_LOAD) {
+		if (StrEqualsIgnoreCase(ext, ".json")) {
```

Comment:
This seems very wide-open. What would happen if we added  some different metadata later that is also stored in a file ending .json?

Reply:
We could invent a file extension, at the cost of scenario creators needing to change file extension after exporting JSON from the spreadsheet used to generate the data. I did experiment with `.geojson` but that's actually a standardized format.

Reply:
Actually, I think you might be mistaken. This callback is only triggered when the user clicks "Load [town data] from file" -- so it does NOT reserve the file extension `.json` for town data. Unless I'm very mistaken, which is always possible. 😄